tokio/src/sync/rwlock/owned_read_guard.rs

```diff
@@ -138,6 +138,32 @@ impl<T: ?Sized, U: ?Sized> OwnedRwLockReadGuard<T, U> {
             resource_span: this.resource_span,
         })
     }
+
+    /// Returns a reference to the original `Arc<RwLock>`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::sync::Arc;
+    /// use tokio::sync::{RwLock, OwnedRwLockReadGuard};
+    ///
+    /// #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+    /// struct Foo(u32);
+    ///
+    /// # #[tokio::main]
+    /// # async fn main() {
+    /// let lock = Arc::new(RwLock::new(Foo(1)));
+    ///
+    /// let guard = lock.clone().read_owned().await;
+    /// assert!(Arc::ptr_eq(&lock, guard.rwlock()));
+    ///
+    /// let guard = OwnedRwLockReadGuard::map(guard, |f| &f.0);
+    /// assert!(Arc::ptr_eq(&lock, guard.rwlock()));
+    /// # }
+    /// ```
+    pub fn rwlock(&self) -> &Arc<RwLock<T>> {
```

Comment:
We shouldn't add new `self` methods to types that implement `Deref`.
```suggestion
    pub fn rwlock(this: &Self) -> &Arc<RwLock<T>> {
```

Reply:
Ah, makes sense!

Reply:
@Darksonn 
Probably I'm not following here. Could you elaborate on why we shouldn’t add new &self methods to types that implement Deref?

Reply:
@mox692 If the `Target` type of the `Deref` implementation (`T` in this case) already has a method `fn rwlock(&self)` then this will break backwards compatibility as instead of calling the target's method it will call the guard's.

Reply:
@r3v2d0g
Got it, make sense. Thanks for the clarification.